app/models/contract.rb

```diff
@@ -62,10 +62,15 @@ def self.issued_by(issuer, *ids)
     where.has { plan_id.in( scope ) }
   end
 
+  # FIXME: this probably shouldn't return `all`, it's dangerous if it is accidentally
+  # used as a separate scope, and not merged with other scopes that filters by account previously
   scope :permitted_for, ->(user) {
-    next all unless user.forbidden_some_services?
+    permitted_services_status = user.permitted_services_status
+    next none if permitted_services_status == :none
 
-    where(service_id: user.member_permission_service_ids)
+    next all if permitted_services_status == :all
+
+    merge(permitted_services_status == :selected ? where(service_id: user.member_permission_service_ids) : {})
```

Comment:
Could you explain what is the actual difference in behavior compared to before. I don't really understand it and I can't properly review if I don't understand this.

Reply:
Well, I'll try...

From the user point of view, the issue was that a freshly created member that has NO permissions defined at all had access to some pages - specifically mapping rules, policies, metrics etc. They were not visible from the menus, but using a direct link was possible.

The main reason was in this scope, together with the cancan definition:
```
can %i[read show edit update], Service, user.accessible_services.where_values_hash
```

So, if the user had no `:services` permission, it was automatically considered that it had access to all services, regardless of whether they had permissions such as `partners`, `plans` or `monitoring`.

Currently, it is considered that the user has access to all services ONLY if in addition to not having `:services` permission there is at least one of the "per-service" permissions (`partners`, `plans` or `monitoring`).

So, a newly created member, that does not have any permissions at all will not be considered as "being able to see all services".

Reply:
> Currently, it is considered that the user has access to all services ONLY if in addition to not having `:services` permission there is at least one of the "per-service" permissions (`partners`, `plans` or `monitoring`).

Do you have a reference for this?

Reply:
Well, it's not that straightforward, you kind of need to follow the whole chain. So, this is what the scope use to say:
```
self.merge(
      account_services.merge(user.forbidden_some_services? ? where(id: user.member_permission_service_ids) : {})
    )
```

`user.forbidden_some_services?` call `!has_access_to_all_services?`, `has_access_to_all_services` calls `!admin_sections.include?(:services)`.

So, if `admin_sections` does **NOT** include `:services` (which happens to a member with no permissions at all), `has_access_to_all_services == true`, `user.forbidden_some_services? == false`, so the scope will return `all`.

Reply:
To clarify, the above is for the `Service.permitted_for` scope, not for the Contract.

But when testing I realised the `Contract.permitted_for` was wrong also.

Reply:
For clarity, I would suggest to refactoring it into something like:
```rb
case permitted_services_status
when :all, :none
  public_send permitted_services_status
when :selected
  where(service_id: user.member_permission_service_ids)
else
  {} # but what else could be the value? doesn't `{}` in this case mean the same as `all`?
end
```

Reply:
Ha, nice point. I hadn't even realized that the values `:all` and `:none` (that `#permitted_services_status` returns) were the same as `all` and `none` scopes in AR :grimacing: 

I'll give it a try